jolt-core/src/jolt/vm/bytecode.rs

```diff
@@ -175,6 +216,44 @@ impl<F: PrimeField> FiveTuplePoly<F> {
             self.imm.evaluate(r),
         ]
     }
+
+    fn from_elf_r1cs(elf: &Vec<ELFRow>, N_SKIP: usize) -> Vec<F> {
+        // Do not pad. 
+        let len = elf.len();
+
+        let mut opcodes = Vec::with_capacity(len);
+        let mut rds = Vec::with_capacity(len);
+        let mut rs1s = Vec::with_capacity(len);
+        let mut rs2s = Vec::with_capacity(len);
+        let mut imms = Vec::with_capacity(len);
+        // TODO: handle circuit flags here and not in prove_r1cs()
+        // let mut circuit_flags = Vec::with_capacity(len * 15);
+
+        for row in elf {
+            opcodes.push(F::from(row.opcode));
+            rds.push(F::from(row.rd));
+            rs1s.push(F::from(row.rs1));
+            rs2s.push(F::from(row.rs2));
+            imms.push(F::from(row.imm));
+            // circuit_flags.push(row.circuit_flags_packed::<F>());
+        }
+
+        // skip the first N_SKIP elements of each vector 
+        let opcodes = opcodes.iter().skip(N_SKIP).cloned().collect::<Vec<_>>();
```

Comment:
Clones are very expensive (especially when the vectors are this large. Let's instead deal with this in the for loop above and never add the first `N_SKIP` elements to the vectors.